src/services/completions.ts

```diff
@@ -914,6 +924,52 @@ namespace ts.Completions {
         }
 
         /**
+         * Aggregates relevant symbols for completion in class declaration
+         * Relevant symbols are stored in the captured 'symbols' variable.
+         *
+         * @returns true if 'symbols' was successfully populated; false otherwise.
+         */
+        function tryGetClassLikeCompletionSymbols(classLikeDeclaration: ClassLikeDeclaration): boolean {
+            // We're looking up possible property names from parent type.
+            isMemberCompletion = true;
+            // Declaring new property/method/accessor
+            isNewIdentifierLocation = true;
+            // Has keywords for class elements
+            hasFilteredClassMemberKeywords = true;
+
+            const baseTypeNode = getClassExtendsHeritageClauseElement(classLikeDeclaration);
+            if (baseTypeNode) {
+                const classElement = contextToken.parent;
+                let classElementModifierFlags = isClassElement(classElement) && getModifierFlags(classElement);
+                // If this is context token is not something we are editing now, consider if this would lead to be modifier
+                if (contextToken.kind === SyntaxKind.Identifier && !isCurrentlyEditingNode(contextToken)) {
+                    switch (contextToken.getText()) {
+                        case "private":
+                            classElementModifierFlags = classElementModifierFlags | ModifierFlags.Private;
+                            break;
+                        case "static":
+                            classElementModifierFlags = classElementModifierFlags | ModifierFlags.Static;
+                            break;
+                    }
+                }
+
+                // No member list for private methods
+                if (!(classElementModifierFlags & ModifierFlags.Private)) {
+                    const baseType = typeChecker.getTypeAtLocation(baseTypeNode);
+                    const typeToGetPropertiesFrom = (classElementModifierFlags & ModifierFlags.Static) ?
+                        typeChecker.getTypeOfSymbolAtLocation(baseType.symbol, classLikeDeclaration) :
+                        baseType;
+
+                    // List of property symbols of base type that are not private
+                    symbols = filter(typeChecker.getPropertiesOfType(typeToGetPropertiesFrom),

```

Comment:
@DanielRosenwasser commented on the issue saying not to do that. Reference https://github.com/Microsoft/TypeScript/issues/7158#issuecomment-188991281